src/udev/udev-builtin-net_id.c

```diff
@@ -342,9 +342,26 @@ static int dev_pci_slot(sd_device *dev, struct netnames *names) {
 
         hotplug_slot_dev = names->pcidev;
         while (hotplug_slot_dev) {
+                int function_id;
+
                 if (sd_device_get_sysname(hotplug_slot_dev, &sysname) < 0)
                         continue;
 
+                /* match slot address to function_id, if available */
+                if (naming_scheme_has(NAMING_SLOT_FUNCTION_ID) &&
+                    sd_device_get_sysattr_value(hotplug_slot_dev, "function_id", &attr) >= 0 &&
+                    safe_atoi(attr, &function_id) >= 0) {
+                        char str[PATH_MAX];
+
+                        if (snprintf_ok(str, sizeof str, "%s/%08x", slots, function_id) &&
+                            access(str, R_OK | X_OK) == 0) {
+                                hotplug_slot = function_id;
+                                domain = 0;
+                        }
```

Comment:
Shouldn't there be at least a debug message here if the check fails? As you said in the other ticket
> If there's a function_id, there also must be a matching slot, otherwise the system is in a inconsistent state.

Reply:
There was no precedent for log messages in here, so I wasn't sure whether this would be acceptable. If nobody objects, I can for sure add one.

Reply:
Yes, please add it. udev had a severe deficiency of log messages in the past, and we're slowly fixing that. New or touched code should log where appropriate.